src/platform/plugins/shared/dashboard/server/index.ts

```diff
@@ -8,13 +8,24 @@
  */
 
 import { PluginInitializerContext, PluginConfigDescriptor } from '@kbn/core/server';
-import { configSchema, ConfigSchema } from './config';
+import { schema, TypeOf } from '@kbn/config-schema';
 
-export const config: PluginConfigDescriptor<ConfigSchema> = {
+export const configSchema = schema.object({
+  allowByValueEmbeddables: schema.boolean({ defaultValue: true }),
+});
+
+export const config: PluginConfigDescriptor<TypeOf<typeof configSchema>> = {
   exposeToBrowser: {
+    /**
+     * this config is unused, but cannot be removed as removing a yml setting is a breaking change.
+     * This can be removed in 10.0. https://github.com/elastic/kibana/issues/221197
+     */
     allowByValueEmbeddables: true,
   },
   schema: configSchema,
+  deprecations: ({ deprecate }) => {
+    return [deprecate('allowByValueEmbeddables', '9.3.0', { level: 'warning' })];
```

Comment:
Why is version 9.3.0?

Reply:
This key is called `removeBy` - I read that as "this warning will tell Kibana administrators to remove this YML config by this minor"

We're planning on removing this in 10.0, so I figured that it might be good to remind any admins that have this setting to remove it ASAP. If this is meant to remind _us_ when to remove the YML config, then it's 10.0. 

A lot of usages of this seem to have "a future version" in that field. I'd be happy to leave it as 9.3, or to switch it to 10.0.

Reply:
why 9.3 instead of the released version of 9.1. 9.3 seems like an odd choice.

Reply:
lets go with 9.1 since its marked as deprecated in that version